apisix/discovery/nacos/init.lua

```diff
@@ -281,11 +283,19 @@ local function fetch_full_registry(premature)
                 up_apps[namespace_id]
                     [group_name][service_info.service_name] = nodes
             end
-            core.table.insert(nodes, {
+
+            local node = {
                 host = host.ip,
                 port = host.port,
                 weight = host.weight or default_weight,
-            })
+            }
+
+            local is_grpc = ngx_re_match(scheme, "grpc", "jo")
```

Comment:
There are only grpc & grpcs, so `scheme == ...` is enough. No need to use regex.